webknossos/webknossos/dataset/_utils/pims_imagej_tiff_reader.py

```diff
@@ -0,0 +1,57 @@
+from os import PathLike
+from pathlib import Path
+from typing import Set
+
+import numpy as np
+
+try:
+    from pims import FramesSequenceND
+except ImportError as e:
+    raise RuntimeError(
+        "Cannot import pims, please install it e.g. using 'webknossos[all]'"
+    ) from e
+
+try:
+    import tifffile
+except ImportError as e:
+    raise RuntimeError(
+        "Cannot import tifffile, please install it e.g. using 'webknossos[tifffile]'"
+    ) from e
+
+
+class PimsImagejTiffReader(FramesSequenceND):
+    @classmethod
+    def class_exts(cls) -> Set[str]:
+        return {"tif", "tiff"}
+
+    # class_priority is used in pims to pick the reader with the highest priority.
+    # Default is 10, and bioformats priority is 2.
+    # See http://soft-matter.github.io/pims/v0.6.1/custom_readers.html#plugging-into-pims-s-open-function
+    class_priority = 20
+
+    def __init__(self, path: PathLike) -> None:
+        super().__init__()
+        path = Path(path)
+        tiff = tifffile.TiffFile(path)
+        assert tiff.is_imagej, f"{path} is not an ImageJ Tiff"
```

Comment:
Will pims use another reader, if this assertion fails?

Reply:
Yes, as long as `__init__` fails, the next compatible reader is used.
See
https://github.com/soft-matter/pims/blob/main/pims/api.py#L199-L205